wp-web-push/lib/delapuente/wp-sw-concoct/class-wp-sw-concoct.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+require_once(__DIR__ . '/class-wp-sw-concoct-router.php');
+require_once(__DIR__ . '/class-wp-sw-concoct-service-worker.php');
+
+class WP_SW_Concoct {
+    const SW_REGISTRAR_SCRIPT = 'wp-sw-concoct-manager';
+
+    private static $instance;
+
+    public static function init() {
+        if (!self::$instance) {
+            self::$instance = new self();
+        }
+    }
+
+    public static function get_manager() {
+        self::init();
+        return self::$instance;
+    }
+
+    private $router;
+
+    private $sw_registrar_script_url;
+
+    private $service_workers;
+
+    private $enabled_sw;
+
+    private function __construct() {
+        $this->router = WP_SW_Concoct_Router::get_router();
+        $this->enabled_sw = array();
+        $this->service_workers = array();
+        $this->setup_sw_registrar_script();
+    }
+
+    public function sw($scope='') {
+        if (!$scope) { $scope = $this->default_scope(); }
+        if (!array_key_exists($scope, $this->service_workers)) {
+            $this->add_new_sw($scope);
+        }
+        return $this->service_workers[$scope];
+    }
+
+    public function sw_js_id($scope='') {
+        if (!$scope) { $scope = $this->default_scope(); }
+        return $scope;
+    }
+
+    public function enable_sw($scope='') {
+        if (!$scope) { $scope = $this->default_scope(); }
+        $this->enabled_sw[] = $scope;
+    }
+
+    public function sw_registrar() {
+        header('Content-Type: application/javascript');
+        $contents = $contents = file_get_contents(__DIR__ . '/lib/js/sw-registrar.js');
```

Comment:
We could make sw-registrar a PHP script, so you don't have to read it with file_get_contents and use str_replace, but you can simply require it.